packages/typescript/src/lib/Typescript.ts

```diff
@@ -123,30 +123,54 @@ export class FederatedTypesPlugin {
 
   private extractTypes() {
     if (this.exposedComponents) {
-      const normalizedFileNames = Object.values(this.exposedComponents)
-        .map((exposed) => {
-          const [rootDir, entry] = exposed.split(/\/(?=[^/]+$)/);
-
+      const exposeSrcToDestMap: Record<string, string> = {};
+      const normalizedFileNames = Object.entries(this.exposedComponents)
+        // './Component': 'path/to/component' -> ['./Component', 'path/to/component']
+        .map(([exposeDest, exposeSrc]) => {
+          const [rootDir, entry] = exposeSrc.split(/\/(?=[^/]+$)/);
           const cwd = this.webpackCompilerOptions.context || process.cwd();
           const normalizedRootDir = path.resolve(cwd, rootDir);
           const filenameWithExt = this.resolveFilenameWithExtension(
             normalizedRootDir,
             entry
           );
-
-          return path.resolve(cwd, rootDir, filenameWithExt);
+          const pathWithExt = path.resolve(cwd, rootDir, filenameWithExt);
```

Comment:
This is not needed as `filenameWithExt` includes the full path resolved to the file.

Reply:
Logging the two, this is what i get
```
{
  filenameWithExt: 'counter.ts',
  pathWithExt: '/Users/haydonlam/Documents/projects/home-fe-test/test/counter.ts'
}
```